### PR TITLE
helm: fix duplicate configmap key for `bpf-lb-sock-terminate-pod-connections`

### PR DESCRIPTION
**Pull request by solidDoWant**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Previously, when both `socketLB.enabled` and `socketLB.terminatePodConnections` were set in the Helm chart, the cilium-configmap would include the `bpf-lb-sock-terminate-pod-connections` twice. This fixes the issue by only emitting the key once.

This issue appears to affect Flux CD HelmReleases, but not Helmfile for some reason. I _think_ Flux renders the Helm chart internally prior to sending the manifests to the API server, whereas Helmfile invokes `helm`, which appears to remove or otherwise merge duplicate keys.

```release-note
helm: fix duplicate configmap key for `bpf-lb-sock-terminate-pod-connections`
```

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 1]
```diff
@@ -727,13 +727,14 @@ data:
...
   bpf-lb-sock: {{ $socketLB.enabled | quote }}
-  bpf-lb-sock-terminate-pod-connections: {{ $socketLB.enabled | quote }}
 {{- end }}
 {{- if hasKey $socketLB "hostNamespaceOnly" }}
   bpf-lb-sock-hostns-only: {{ $socketLB.hostNamespaceOnly | quote }}
 {{- end }}
 {{- if hasKey $socketLB "terminatePodConnections" }}
   bpf-lb-sock-terminate-pod-connections: {{ $socketLB.terminatePodConnections | quote }}
+{{- else }}
+  bpf-lb-sock-terminate-pod-connections: {{ $socketLB.enabled | quote }}
```
Comment (solidDoWant) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml:737]: The values file indicates that this should default to `enabled`, but the former logic defaults it to `$socketLB.enabled`. Not sure which is desired.

**Review by youngnick — APPROVED** [review]
LGTM from sig-k8s.

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 1 above)
Comment (gandro) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:737]: It is also auto-enabled if hostServices are enabled, see https://github.com/cilium/cilium/blob/19de3f774778221303c16c7b1fc41befaaadc797/install/kubernetes/cilium/templates/cilium-configmap.yaml#L74

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 2]
```diff
@@ -727,13 +727,14 @@ data:
...
 {{- if hasKey $socketLB "enabled" }}
   bpf-lb-sock: {{ $socketLB.enabled | quote }}
-  bpf-lb-sock-terminate-pod-connections: {{ $socketLB.enabled | quote }}
 {{- end }}
 {{- if hasKey $socketLB "hostNamespaceOnly" }}
   bpf-lb-sock-hostns-only: {{ $socketLB.hostNamespaceOnly | quote }}
 {{- end }}
 {{- if hasKey $socketLB "terminatePodConnections" }}
   bpf-lb-sock-terminate-pod-connections: {{ $socketLB.terminatePodConnections | quote }}
+{{- else }}
```
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]: Should we replace this `else` statement here with an `else if hasKey $socketLB "enabled"` to fully preserve the previous behavior?

**Review by gandro — COMMENTED** [review]
Looks good overall. One minor question

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 2 above)
Comment (solidDoWant) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml]: Sounds good to me - added.

**Review by gandro — COMMENTED** [review]
Thanks for the change! Could you squash your commits into one with `git rebase`? Other than that this PR should then be good to go.

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 2 above)
Comment (gandro) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml]: Thank you!

**Comment by solidDoWant** [comment]
Sure - done.

**Review by gandro — APPROVED** [review]

**Pull request merged**